docs/zephyr_02_functional_requirements.sdoc

```diff
@@ -1379,6 +1364,158 @@ As a Zephyr OS user I want to... ???
 
 [/SECTION]
 
+[SECTION]
+TITLE: Thread Synchronization
+
+[REQUIREMENT]
+UID: ZEP-SEMAPHORE-001
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Semaphore
+TITLE: Counting Semaphore Creation At Build Time
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to create semaphores at build time.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-99
+
+[REQUIREMENT]
+UID: ZEP-SEMAPHORE-002
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Semaphore
+TITLE: Counting Semaphore Creation At Run Time
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to create semaphores at runtime.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-99
+
+[REQUIREMENT]
+UID: ZEP-SEMAPHORE-003
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Semaphore
+TITLE: Semaphore Initialization
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to initialize semaphore values to regulate the number of tasks or resources concurrently accessing a shared resource.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-99
+
+[REQUIREMENT]
+UID: ZEP-SEMAPHORE-004
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Semaphore
+TITLE: Semaphore Initialization Options
+STATEMENT: >>>
+When initializing a counting semaphore, the Zephyr RTOS shall accept options specifying initial semaphore value and the maximum permitted count a semaphore can have.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-99
+
+[REQUIREMENT]
+UID: ZEP-SEMAPHORE-010
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Semaphore
+TITLE: Semaphore Initialization Option Validation
+STATEMENT: >>>
+When initializing a counting semaphore, where the initial semaphore value and the maximum permitted count a semaphore are invalid,
+the Zephyr RTOS shall return an error.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-99
+
+[REQUIREMENT]
+UID: ZEP-SEMAPHORE-005
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Semaphore
+TITLE: Semaphore acquisition
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism allowing tasks to take (acquire) semaphores, ensuring atomic and thread-safe operations for
+semaphore acquisition.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-99
+
+[REQUIREMENT]
+UID: ZEP-SEMAPHORE-011
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Semaphore
+TITLE: Semaphore release
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism allowing tasks to give (release) semaphores, ensuring atomic and thread-safe operations for
+semaphore release.
+<<<
```

Comment:
", ensuring ..." reads like a justification. Justifications need to be dropped from the statement. If "atomic" is a part of the required behavior then it needs to be in the paired-down statement.